src-ext/TypeScript/README.md

```diff
@@ -0,0 +1,50 @@
+# TypeScript part of ggvega
```

Comment:
When writing something, I find it useful to keep in mind my audience. For this document, I think the audience is:

- people running the GSOC program, to make sure that we made good use of their resources.
- Ian, Haley, and Wenyu, so that we can remember what you did and why you did it.

These are things I will keep in the back of my mind when writing. 

For an introduction, I like to start with the words: "The purpose of this document is". Sometimes, I will change those words later, sometimes I will not. Using those words in the draft helps me to start the document with the topics I know I will need to focus on. 

For me, here are some things I think are included in the purpose (there are probably other things, too):

- discuss the extensibility of the framework, for example I was always asking how we can make it simpler to be able to add another Geom/Stat/Scale/etc.
- discuss the design choices and compromises we made
  - example: why TypeScript? At the root is that we wanted to base our code on Vega-Lite schema, and we needed an environment where we can "do something" with the schema.
  - example: is the ES5/quicktype issue. A summary will be good here, you can provide a link to your other report for more details.
- to document what each of the TS libraries does (ggschema and ggvega), how our design choices were implemented.
-  I also think this is a good place to document "how to update the code when Vega-Lite updates its schema"